server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1462,6 +1462,25 @@ final boolean tryRenewSyncCommit() {
         return renewed;
     }
 
+    @Override
+    public boolean shouldFlush() {
+        if (translog.shouldFlush() == false) {
+            return false;
+        }
+        /*
+         * We should only flush ony if the shouldFlush condition can become false after flushing. This condition will change if:
+         * 1. The min translog gen of the next commit points to a different translog gen than the last commit
```

Comment:
I think this deserves a comment why we don't take the IW#hasUncommittedChanges() into account.

Reply:
should we call `ensureOpen()` here as well?

Reply:
Done